### PR TITLE
Fix committee and indexes

### PR DESCRIPTION
**Pull request by kderme**
# Description

Fixes https://github.com/IntersectMBO/cardano-db-sync/issues/1735. Also fixes the indexes
It also cherry-picks some forgotten commits from release 13.2.0.2

# Checklist

- [ ] Commit sequence broadly makes sense
- [x] Commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] Any changes are noted in the [changelog](https://github.com/IntersectMBO/cardano-db-sync/blob/master/cardano-db-sync/CHANGELOG.md)
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) on version 0.10.1.0 (which can be run with `scripts/fourmolize.sh`)
- [ ] Self-reviewed the diff

# Migrations

- [ ] The pr causes a [breaking change](https://github.com/IntersectMBO/cardano-db-sync/blob/master/doc/migrations.md) of type a,b or c
- [ ] If there is a breaking change, the pr includes a database migration and/or a fix process for old values, so that upgrade is possible
- [ ] Resyncing and running the migrations provided will result in the same database semantically

If there is a breaking change, especially a big one, please add a justification here. Please elaborate
more what the migration achieves, what it cannot achieve or why a migration is not possible.

### CODE REVIEW

File: cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Query.hs [hunk 1]
```diff
@@ -82,41 +44,3 @@ resolveInputTxOutIdValue txIn =
...
-    -- Need to order by DelegationSlotNo descending for correct behavior when there are two
-    -- or more delegation certificates in a single epoch.
-    orderBy [desc (blk ^. BlockSlotNo)]
-    limit 1
-    pure (sr ^. StakeRegistrationAddrId)
-  pure $ unValue <$> listToMaybe res
-
--- Check if there are other PoolUpdates in the same blocks for the same pool
-queryPoolUpdateByBlock :: MonadIO m => BlockId -> PoolHashId -> ReaderT SqlBackend m Bool
-queryPoolUpdateByBlock blkId poolHashId = do
```
Comment (Cmdv) [inline cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Query.hs:109]: 👌 I did find it confusing that we had db queries inside of cardano-db-sync but was too scared to move 😅

**Review by Cmdv — CHANGES_REQUESTED** [review]
LGTM just a simple export might need removing

**Comment by sgillespie** [comment]
Looks like you have some tests to update:

```
force-indexes
        check force-index adds indexes:                                          FAIL (51.21s)
          test/Test/Cardano/Db/Mock/Validate.hs:153:
          unexpected number of indexes: got 171 expected 172
          Use -p '/Conway unit tests.Command Line Arguments.force-indexes.check force-index adds indexes/' to rerun this test only.
        check no force-index doesn't add indexes:                                FAIL (51.22s)
          test/Test/Cardano/Db/Mock/Validate.hs:153:
          unexpected number of indexes: got 108 expected 107
          Use -p '/Conway unit tests.Command Line Arguments.force-indexes.check no force-index doesn'\''t add indexes/' to rerun this test only.
```

File: cardano-db-sync/src/Cardano/DbSync/Ledger/Types.hs [hunk 2]
```diff
@@ -143,7 +142,7 @@ data ApplyResult = ApplyResult
   , apSlotDetails :: !SlotDetails
   , apStakeSlice :: !Generic.StakeSliceRes
   , apEvents :: ![LedgerEvent]
-  , apCommittee :: !(Maybe (StrictMaybe (Committee StandardConway)))
+  , apGovActionState :: !(Maybe (ConwayGovState StandardConway))
```
Comment (sgillespie) [inline cardano-db-sync/src/Cardano/DbSync/Ledger/Types.hs:145]: :ok_hand: This will come in handy for testing, too

**Review by Cmdv — COMMENTED** [review]
LGTM just a couple of little things. Also do we need documentation for `EnableFutureGenesis` in the docs?

**Comment by kderme** [comment]
Done at https://github.com/IntersectMBO/cardano-db-sync/pull/1744

**Pull request closed without merging**